src/monogame/include/csharp_MGG.h

```diff
@@ -0,0 +1,26 @@
+//
+// This code is auto generated, don't modify it by hand.
+// To regenerate it run: Tools/MonoGame.Generator.CTypes
+//
+
+#pragma once
+
+#include "csharp_common.h"
+#include "csharp_enums.h"
+#include "csharp_structs.h"
+
+
+struct MGG_GraphicsDevice;
```

Comment:
The idea here is the C++ implementation will define the internals of these pre-declared structures.

For example one for DX could look like:

```
struct MGG_GraphicsDevice
{
	ID3D12Device* _device;
	ID3D12GraphicsCommandList* _commandList;
	ID3D12CommandQueue* _commandQueue;
       // etc!
}
```

This would be how native platforms pass along its data across pinvokes.

The C# just passes pointers and the C++ is responsible for allocation and freeing of these types.